python/lsst/analysis/drp/makeQaTractTables.py

```diff
@@ -0,0 +1,353 @@
+                 "base_PsfFlux", "base_CircularApertureFlux_9_0_instFlux", "base_CircularApertureFlux_12_0",
+                 "base_CircularApertureFlux_25_0", "ext_photometryKron_KronFlux", "modelfit_CModel",
+                 "base_Sdss", "slot_Centroid", "slot_Shape", "ext_shapeHSM_HsmSourceMoments_",
+                 "ext_shapeHSM_HsmPsfMoments_", "ext_shapeHSM_HsmShapeRegauss_", "base_Footprint",
+                 "base_FPPosition", "base_ClassificationExtendedness", "parent", "detect", "deblend_nChild",
+                 "base_Blendedness_abs", "base_Blendedness_flag", "base_InputCount",
+                 "merge_peak_sky", "merge_measurement", "calib", "sky_source"],
+        doc=("List of \"startswith\" strings of column names to load from deepCoadd_obj parquet table. "
+             "All columns that start with one of these strings will be loaded UNLESS the full column "
+             "name contains one of the strings listed in the notInColumnStrList config.")
+    )
+    notInColStrList = pexConfig.ListField(
+        dtype=str,
+        default=["flag_bad", "flag_no", "missingDetector_flag", "_region_", "Truncated", "_radius",
+                 "_bad_", "initial", "_exp_", "_dev_", "fracDev", "objective", "SdssCentroid_flag_",
+                 "SdssShape_flag_u", "SdssShape_flag_m", "_Cov", "_child_", "_parent_"],
+        doc=("List of substrings to select against when creating list of columns to load from the "
+             "deepCoadd_obj parquet table.")
+    )
+
+
+class BaseMakeQaTractTablesTask(pipeBase.PipelineTask):
+    """Read in {coaddName}Coadd_obj tables and aggregate QA columns by tract.
+
+    For a given tract, read in all of the per-patch {coaddName}Coadd_obj
+    tables, select on a subset of columns that are useful for QA analyses and
+    plotting routines, aggregate these by tract and persist them as datasets.
+    A separate table is created for each band.  These serve the purpose of
+    providing columns that are useful for QA but are not (yet?) included in the
+    DPDD-based per-tract objectTable_tract tables.
+
+    For ease of use with along with the objectTable_tract tables (e.g. being
+    able to use a single boolean expression to filter on both), the columns of
+    these new tables are sorted in the same order as in the objectTable_tract
+    tables and, since we don't want to make assumptions on that ordering, we
+    load in the table itself to do the sorting.
+
+    Subclasses must implement their own collateTable() method to collect the
+    relevant columns from the input tables.
+    """
+
+    ConfigClass = BaseMakeQaTractTablesTaskConfig
+    _DefaultName = "baseMakeQaTractTablesTask"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        inputs = butlerQC.get(inputRefs)
+        inputs["band"] = butlerQC.quantum.outputs["qaTractTable_" + self.MEASUREMENT_TYPE][0].dataId["band"]
```

Comment:
```suggestion
        inputs["band"] = quantum.dataId["band"]
```

This is not just simpler; it's also strongly discouraged to use the actual dataset type names (like "qaTractTable_unforced") instead of the internal names (i.e. just "qaTractTable") inside a `PipelineTask`, as that prevents users from reconfiguring the task to write to or read from a different dataset type later.

It also lets you remove the `MEASUREMENT_TYPE` attributes from the derived classes, as this is the only place they now seem to be used.

Reply:
Heh...I was kinda waiting for you to flag this one.  The problem with your suggestion (and why I didn't do it in the first place) is that this may be a rare case where there is no `band` dimension in my input connections since neither the `_obj` or `_objTable_tract` inputs have `band` as part of their `dataId`.  So, for example, the `dataId` I get from the above is:
```
print(butlerQC.quantum.dataId)
{skymap: 'hsc_rings_v1', tract: 9813}
```
However, inspired by your kick (along with the `cT.Output` shuffle), I looked further into the `outputRefs` object, and I think I've simplified this down to:
```
inputs["band"] = outputRefs.qaTractTable.dataId["band"]
```
which I hope doesn't fall under the "strongly discouraged" category!

Reply:
Glad you caught my mistake, and your fix is exactly the right one!  (I was actually going to suggest that first, but I figured `quantum.dataId` would be even simpler.)